app/components/searchable-select/component.js

```diff
@@ -0,0 +1,287 @@
+/**
+ * @fileOverview
+ * @name component<searchable-select>
+ *
+ * Fetures:
+ * 1. options search/filter
+ * 2. grouping
+ * 3. show icons for unGroupedContent options
+ * 4. `arrow-up` & `arrow-down` keys to navigate through options
+ * 5. `return` key to select the current active option
+ * 6. esc to cancel
+ *
+ * Option data structure:
+ * {
+ *  label: string,
+ *  value: string,
+ *  group: string,   // Optional, which group/category this option belong to.
+ *  imgUrl: string,  // Optional, whether to display a image for this option, unGrouped options only.
+ * }
+ *
+**/
+
+import Ember from 'ember';
+
+const KEY_CODES  = {
+  return: 13,
+  esc: 27,
+  up: 38,
+  down: 40,
+}
+
+export default Ember.Component.extend({
+  classNames: ['searchable-select'],
+  classNameBindings: ['class'],
+
+  intl: Ember.inject.service(),
+
+  // input
+  value: null,
+  prefix: null,
+  suffix: null,
+  prompt: null,
+  class: null,
+  // If need to catch the group changes, you can pass a group prop in.
+  group: null,
+  content: [],
```

Comment:
`content: null`.  The user is going to overwrite it anyway so it doesn't matter much here, but if they don't then _every_ instance of this component will share the _same_ array when written like this and cause weird interaction between instances.

If you ned to set the default for a field to an object or array, set it in `init()` so each instance gets it's own copy.